### DISCUSSION TITLE
getRequestSources with source type observable

### DISCUSSION THREAD

**Discussion by chfourie** [state-adapt/state-adapt#35] [Ideas]
Idea for addition to API - something like: 

function getRequestSources<Payload>(obs$: Source<Payload>)...

... where the type of the source is used as type prefix rather than having to explicitly pass in a type

**Comment by mfp22** [comment]
This sounded good at first, but it would create a imperative pattern where the source exists outside of what is pushing to it. If there's an example where it would be nice to have I'd like to understand more about it.
